src/murfey/server/api/clem.py

```diff
@@ -0,0 +1,665 @@
+    """
+
+    # Validate that parameters are provided correctly
+    if file_path is None and series_name is None:
+        raise ValueError(
+            "One of either 'file_path' or 'series_name' has to be provided"
+        )
+    if file_path is not None and series_name is not None:
+        raise ValueError("Only one of 'file_path' or 'series_name' should be provided")
+
+    # Validate file path if provided
+    if file_path is not None:
+        try:
+            file_path = validate_and_sanitise(file_path)
+        except Exception:
+            raise Exception
+
+    # Validate series name to use
+    if series_name is not None:
+        if bool(re.fullmatch(r"^[\w\s\.\-]+$", series_name)) is False:
+            raise ValueError("One or more characters in the string are not permitted")
+
+    # Return database entry if it exists
+    try:
+        db_entry = (
+            db.exec(
+                select(table)
+                .where(table.session_id == session_id)
+                .where(table.file_path == str(file_path))
+            ).one()
+            if file_path is not None
+            else db.exec(
+                select(table)
+                .where(table.session_id == session_id)
+                .where(table.series_name == series_name)
+            ).one()
+        )
+    # Create and register new entry if not present
+    except NoResultFound:
+        db_entry = (
+            table(
+                file_path=str(file_path),
+                session_id=session_id,
+            )
+            if file_path is not None
+            else table(
+                series_name=series_name,
+                session_id=session_id,
+            )
+        )
+        db.add(db_entry)
+        db.commit()
+        db.refresh(db_entry)
+    except Exception:
+        raise Exception
+
+    return db_entry
+
+
+"""
+API ENDPOINTS FOR FILE REGISTRATION
+"""
+
+
+@router.post("/sessions/{session_id}/clem/lif_files")
+def register_lif_file(
+    lif_file: Path,
+    session_id: int,
+    master_metadata: Optional[Path] = None,
+    child_metadata: list[Path] = [],
+    child_series: list[str] = [],
+    child_stacks: list[Path] = [],
+    db: Session = murfey_db,
+):
+    # Return or register the LIF file entry
+    try:
+        clem_lif_file: CLEMLIFFile = get_db_entry(
+            db=db,
+            table=CLEMLIFFile,
+            session_id=session_id,
+            file_path=lif_file,
+        )
+    except Exception:
+        logger.error(traceback.format_exc())
+        return False
+
+    # Add metadata information if provided
+    if master_metadata is not None:
+        try:
+            master_metadata = validate_and_sanitise(master_metadata)
+            clem_lif_file.master_metadata = str(master_metadata)
+        except Exception:
+            logger.warning(traceback.format_exc())
+
+    # Register child metadata if provided
+    if len(child_metadata) > 0:
```

Comment:
probably more pythonic to just use 
`if child_metadata:`
here

Reply:
in fact you probably don't need that at all. if the length is 0 the for loop just won't do anything